src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs

```diff
@@ -1620,6 +1620,78 @@ public void AddTimeSeriesNameToMetadata(DocumentsOperationContext ctx, string do
             }
         }
 
+        private static string EnsureOriginalName(DocumentsOperationContext ctx, string docId, string tsName)
+        {
+            var parts = tsName.Split(TimeSeriesConfiguration.TimeSeriesRollupSeparator);
```

Comment:
Can we have '@' in the name of TS?

Reply:
https://issues.hibernatingrhinos.com/issue/RavenDB-14800